api/v1beta1/nutanixcluster_validate_test.go

```diff
@@ -102,3 +106,105 @@ func TestNutanixClusterOpenAPIValidationPCAddress(t *testing.T) {
 		})
 	}
 }
+
+func TestNutanixClusterFailureDomainValidation(t *testing.T) {
```

Comment:
I think it's better to add test cases to the controller/nutanixcluster_controller_test.go, to validate setting of the FD fields (not allow to set both "failureDomains" and "controlPlaneFailureDomains" simultaneously) by creating the actual NuanixCluster object with envtest and check the expected error throw by the apiserver. See example of https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/main/controllers/nutanixfailuredomain_controller_test.go testcase "Update an NutanixFailureDomain Configuration" (L122-134).

Reply:
Fair point. That is indeed a more robust test.